### PR TITLE
[WIP] Add database module

### PR DESCRIPTION
**Pull request by jesusgn90**
**Connected to** https://github.com/islos-efe-eme/gta-tools/issues/31

:warning: this is not ready yet and this introduces dependency changes

### CODE REVIEW

File: db/src/index.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
+import { config } from 'dotenv'
+import low, { LowdbSync, AdapterSync } from 'lowdb'
+import FileSync from 'lowdb/adapters/FileSync'
+import { GenericMap } from './models/generic-map'
+import { User, CreateUser } from './models/user'
+import { ERROR_MESSAGES } from './helpers/errorMessages'
+import { MISC } from './helpers/misc'
+// Load env variables
+config()
```
Comment (jesusgn90) [inline db/src/index.ts]: Do we really need .env for this module? If used as standalone module, I see nothing to be in .env file so far

File: db/src/index.d.ts [hunk 2]
```diff
@@ -0,0 +1,20 @@
+import { LowdbSync, AdapterSync } from 'lowdb'
+
+import { GenericMap, User } from './models'
+
+import { CreateUser } from './models/user'
+
+declare class GTADatabase {
```
Comment (jesusgn90) [inline db/src/index.d.ts]: This is not fine and we should rely on dts-gen or a similar tool to create the types definition file on the fly

File: db/package.json [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+  "author": "@jesusgn90",
+  "license": "GPL-3.0",
+  "bugs": {
+    "url": "https://github.com/islos-efe-eme/gta-tools/issues"
+  },
+  "homepage": "https://github.com/islos-efe-eme/gta-tools#readme",
+  "dependencies": {
+    "dotenv": "^8.2.0",
+    "lowdb": "^1.0.0",
+    "moment": "^2.24.0"
```
Comment (jesusgn90) [inline db/package.json]: Are we using "moment" at all???

File: db/package.json [hunk 4]
```diff
@@ -0,0 +1,51 @@
...
+  },
+  "devDependencies": {
```
Comment (jesusgn90) [inline db/package.json]: Tackle https://github.com/islos-efe-eme/gta-tools/issues/38 too!

File: db/package.json [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    "eslint": "^6.8.0",
+    "eslint-plugin-eslint-comments": "^3.1.2",
+    "eslint-plugin-eslint-plugin": "^2.2.1",
+    "eslint-plugin-import": "^2.20.0",
+    "eslint-plugin-jest": "^23.6.0",
+    "jest": "^25.1.0",
+    "jest-junit": "^10.0.0",
+    "rimraf": "^3.0.1",
+    "ts-jest": "^25.0.0",
+    "typescript": "^3.8.2"
```
Comment (crolopez) [inline db/package.json]: Use 3.8.3, please.

**Pull request closed without merging**